query-engine/core/src/schema_builder/input_type_builder/create_input_type_extension.rs

```diff
@@ -143,8 +137,13 @@ pub trait CreateInputTypeBuilderExtension<'a>: InputTypeBuilderBase<'a> {
                         }
                     };
 
+                    let all_required_scalar_fields_have_defaults = rf
+                        .linking_fields()
+                        .scalar_fields()
+                        .all(|scalar_field| scalar_field.default_value.is_some());
```

Comment:
Do we care about evaluating this only if the field is required, or can the small overhead be ignored for readability in this context?

Reply:
Readability > performance here, because this is build once on startup.